evm/src/generation/state.rs

```diff
@@ -167,6 +172,27 @@ impl<F: Field> GenerationState<F> {
             .map(|i| stack_peek(self, i).unwrap())
             .collect()
     }
+
+    /// Clone everything but the traces
+    pub(crate) fn soft_clone(&self) -> GenerationState<F> {
+        Self {
+            inputs: self.inputs.clone(),
+            registers: self.registers.clone(),
+            memory: self.memory.clone(),
+            traces: Traces::default(),
+            rlp_prover_inputs: self.rlp_prover_inputs.clone(),
+            state_key_to_address: self.state_key_to_address.clone(),
+            bignum_modmul_result_limbs: self.bignum_modmul_result_limbs.clone(),
+            withdrawal_prover_inputs: self.withdrawal_prover_inputs.clone(),
+            trie_root_ptrs: TrieRootPtrs {
+                state_root_ptr: 0,
+                txn_root_ptr: 0,
+                receipt_root_ptr: 0,
+            },
+            last_jumpdest_address: 0,
+            jumpdest_addresses: None,
+        }
+    }
```

Comment:
I need to simulate the cpu from the current state and, after the simulation finishes, continue with the execution of. I don't know how to do it without cloning all these values

Reply:
Isn't it possible to mark a checkpoint and rollback to it once the simulation is done?

Reply:
I tried with the checkpoint and rollback but it fails as long as the memory is modified during the simulation. It works if I manually clone the memory and I restore it back after the rollback

Reply:
Hmm, may help a bit still but `memory` was my biggest concern here with cloning. In any case, I'd need to do more profiling, because I think the search algorithm is also the source of the non-negligible slowdown I noticed when running tests.

Reply:
What are the pieces that are particularly heavy here? Are there any standouts? If there are just a few that cause this to be costly perhaps we could focus on avoiding a clone on just those?

Reply:
Yeah, I also think we'd still benefit from checkpointing + rollback (apart from `memory` which needs cloning) would still be beneficial (although `memory` is the heaviest piece here).
On the other hand it's not clear yet where the runtime overhead is located, I don't think it's only there, but maybe instead in the search algorithm that's not really optimal. Will do a deeper look.